wagtail/core/blocks/base.py

```diff
@@ -73,8 +73,21 @@ def all_blocks(self):
 
     def all_media(self):
         media = forms.Media()
+
+        # In cases where the same block definition appears multiple times within different
+        # container blocks (e.g. a RichTextBlock appearing at the top level of a StreamField
+        # as well as both sides of a StructBlock for producing two-column layouts), we will
+        # encounter identical media declarations. Adding these to the final combined media
+        # declaration would be redundant add processing time when determining the final media
+        # ordering. To avoid this, we keep a cache of previously-seen declarations and only
+        # add unique ones.
+        media_cache = set()
+
         for block in self.all_blocks():
-            media += block.media
+            key = block.media.__repr__()
```

Comment:
There was a question during the core meeting whether using repr is actually unique, otherwise js files may not be loaded.

StreamBlocks use forms.Media, and looking at https://github.com/django/django/blob/master/django/forms/widgets.py#L60, I believe the key will be unique to the combination of css/js a block uses. Moreover, if other blocks have the same css/js combination, we have another win here

Reply:
The `repr` method wraps the `repr` output of the css dict and js list, which both output their elements in full regardless of length (as confirmed with `repr(['foo'] * 1000)`, `repr({a: 'foo' for a in range(1, 1000)})`) so it doesn't look like this is a problem.